src/Connection.php

```diff
@@ -1511,6 +1514,24 @@ public function getWrappedConnection()
         return $this->_conn;
     }
 
+    /**
+     * @return resource|object
+     */
+    public function getNativeConnection()
+    {
+        $this->connect();
+
+        assert($this->_conn !== null);
+        if (! method_exists($this->_conn, 'getNativeConnection')) {
```

Comment:
Would it make sense to introduce a deprecated interface and implement it where possible? In the next major, it will be merged with the driver-level `Connection`.

Reply:
This will make it hard to implement a middleware in a third-party library that is compatible with DBAL 3.2, 3.3 and 4.0 at the same time. That middleware would have to implement an interface that does not exist in all versions.

Reply:
It might be easier to backport the new interface to `3.2.x` and make the code better typed but I don't have a strong preference. The reasoning above makes sense.

Reply:
Thinking about it, the introduction of the abstract middleware classes has opened up the path for the deprecated interface idea, I guess. A middleware would conditionally implement that interface via the abstract class.

Reply:
But isn't the abstract class only available in `3.3.x` as well?

Reply:
True, but we would be able to build middlewares on 3.3 that are forward compatible with 4.0. That would be sufficient for me at least.